include/opw_kinematics/opw_kinematics_impl.h

```diff
@@ -129,70 +132,80 @@ void inverse(const Parameters<T>& params, const Transform<T>& pose, T* out) noex
+
+  Solution<T>& out2 = solutions[2];
+  out2[0] = (theta1_ii + params.offsets[0]) * params.sign_corrections[0];
+  out2[1] = (theta2_iii + params.offsets[1]) * params.sign_corrections[1];
+  out2[2] = (theta3_iii + params.offsets[2]) * params.sign_corrections[2];
+  out2[3] = (theta4_iii + params.offsets[3]) * params.sign_corrections[3];
+  out2[4] = (theta5_iii + params.offsets[4]) * params.sign_corrections[4];
+  out2[5] = (theta6_iii + params.offsets[5]) * params.sign_corrections[5];
+
+  Solution<T>& out3 = solutions[3];
+  out3[0] = (theta1_ii + params.offsets[0]) * params.sign_corrections[0];
+  out3[1] = (theta2_iv + params.offsets[1]) * params.sign_corrections[1];
+  out3[2] = (theta3_iv + params.offsets[2]) * params.sign_corrections[2];
+  out3[3] = (theta4_iv + params.offsets[3]) * params.sign_corrections[3];
+  out3[4] = (theta5_iv + params.offsets[4]) * params.sign_corrections[4];
+  out3[5] = (theta6_iv + params.offsets[5]) * params.sign_corrections[5];
+
+  Solution<T>& out4 = solutions[4];
+  out4[0] = (theta1_i + params.offsets[0]) * params.sign_corrections[0];
+  out4[1] = (theta2_i + params.offsets[1]) * params.sign_corrections[1];
+  out4[2] = (theta3_i + params.offsets[2]) * params.sign_corrections[2];
+  out4[3] = (theta4_v + params.offsets[3]) * params.sign_corrections[3];
+  out4[4] = (theta5_v + params.offsets[4]) * params.sign_corrections[4];
+  out4[5] = (theta6_v + params.offsets[5]) * params.sign_corrections[5];
+
+  Solution<T>& out5 = solutions[5];
+  out5[0] = (theta1_i + params.offsets[0]) * params.sign_corrections[0];
+  out5[1] = (theta2_ii + params.offsets[1]) * params.sign_corrections[1];
+  out5[2] = (theta3_ii + params.offsets[2]) * params.sign_corrections[2];
+  out5[3] = (theta4_vi + params.offsets[3]) * params.sign_corrections[3];
+  out5[4] = (theta5_vi + params.offsets[4]) * params.sign_corrections[4];
+  out5[5] = (theta6_vi + params.offsets[5]) * params.sign_corrections[5];
+
+  Solution<T>& out6 = solutions[6];
+  out6[0] = (theta1_ii + params.offsets[0]) * params.sign_corrections[0];
+  out6[1] = (theta2_iii + params.offsets[1]) * params.sign_corrections[1];
+  out6[2] = (theta3_iii + params.offsets[2]) * params.sign_corrections[2];
+  out6[3] = (theta4_vii + params.offsets[3]) * params.sign_corrections[3];
+  out6[4] = (theta5_vii + params.offsets[4]) * params.sign_corrections[4];
+  out6[5] = (theta6_vii + params.offsets[5]) * params.sign_corrections[5];
+
+  Solution<T>& out7 = solutions[7];
+  out7[0] = (theta1_ii + params.offsets[0]) * params.sign_corrections[0];
+  out7[1] = (theta2_iv + params.offsets[1]) * params.sign_corrections[1];
+  out7[2] = (theta3_iv + params.offsets[2]) * params.sign_corrections[2];
+  out7[3] = (theta4_viii + params.offsets[3]) * params.sign_corrections[3];
+  out7[4] = (theta5_viii + params.offsets[4]) * params.sign_corrections[4];
+  out7[5] = (theta6_viii + params.offsets[5]) * params.sign_corrections[5];
```

Comment:
You might consider using Eigen to vectorize these operations instead. It would save a lot of space and eliminate the need for hard coding the indices: 

```c++
Eigen::Map<Eigen::Array<T, 6, 1>> offsets(params.offsets.data());
Eigen::Map<Eigen::Array<T, 6, 1>> signs(params.sign_corrections.data());

// Create a theta array
Eigen::Array<double, 6, 8> theta;
theta << theta_i, theta_i, theta_i, theta_i, theta_i, theta_i,
    theta_i, theta_ii, theta_ii, theta_ii, theta_ii, theta_ii,
    ...,
    ...,
    ...,
    ...,
    ...,
    ...;

// Perform the computation
Eigen::Array<double, 6, 8> output = (theta.colwise() + offsets).colwise() * signs;

// Save into Solutions object
Solutions solutions;
for(std::size_t i = 0; i < 8; ++i)
{
  ...
}

```

Reply:
Isn't the output fixed-size as well? You should be able to `Eigen::Map` that as well.

Could remove the need for the `for`-loop I believe.

Reply:
Possibly; I'm not sure how an array of arrays stores its data pointer internally. Should be easy enough to test though

Reply:
I agree, and thought about doing this but I believe these changes are outside the scope of this PR.

Reply:
Also there is a throughput test which I have ran with master, switch to array and then vectorized commits. There is not noticeable difference.  If anything the vectorized version is about 0.01 us faster on average.

Reply:
>Isn't the output fixed-size as well? You should be able to Eigen::Map that as well.

@gavanderhoorn It was.